dojo/tools/intsights/parser.py

```diff
@@ -0,0 +1,164 @@
+                alert['status'] = alert.pop('Status')
+                alert['alert_link'] = alert.pop('Alert Link')
+                alert.pop('Assignees')
+                alert.pop('Remediation')
+                alert.pop('Closed Reason')
+                alert.pop('Rating')
+                for key in required_keys:
+                    if not alert[key]:
+                        invalid_alerts.append(alert)
+
+                if alert not in invalid_alerts:
+                    alerts.append(alert)
+        else:
+            self._LOGGER.error('The CSV file has one or more missing or unexpected header values')
+
+        return alerts
+
+    def get_findings(self, file, test):
+        duplicates = dict()
+
+        self._LOGGER.error(type(file))
+
+        if file.name.lower().endswith('.json'):
+            alerts = self._parse_json(file, )
+        elif file.name.lower().endswith('.csv'):
+            alerts = self._parse_csv(file)
+        else:
+            raise ValueError('IntSights report contains errors: Unknown File Format')
+
+        if not alerts:
+            raise ValueError('IntSights report contains errors: No vulnerabilities were found in the data provided')
+
+        for alert in alerts:
+            dupe_key = alert['alert_id']
+
+            if dupe_key in duplicates:
+                alert = duplicates[dupe_key]
+                duplicates[dupe_key] = alert
+            else:
+                duplicates[dupe_key] = True
+
+                alert = Finding(title=alert['title'],
+                                test=test,
+                                active=False if alert['status'] == 'Closed' else True,
+                                verified=True,
+                                description=f'{alert["description"]}'
+                                              f'\n\n----'
+                                              f'\r\n**Date Found**: {alert["report_date"]}'
+                                              f'\n\n----'
+                                              f'\r\n**Type**: {alert["type"]}'
+                                              f'\r\n**Source**: {alert["source_url"]}'
+                                              f'\r\n**Source Date**: {alert["source_date"]}'
+                                              f'\r\n**Source Network Type**: {alert["network_type"]}'
+                                              f'\n\n----'
+                                              f'\r\n**Assets Affected**: {alert["assets"]}'
+                                              f'\n\n----'
+                                              f'\r\n**Alert Link**: {alert["alert_link"]}',
+                                severity=alert['severity'],
+                                references=alert["alert_link"],
+                                static_finding=False,
+                                dynamic_finding=True,
+                                unique_id_from_tool=alert['alert_id'])
```

Comment:
Could you move this part outside the 'if/then' ? better to manage description with duplicates

Reply:
Done!

Reply:
IIRC this was also from the cookie cutter template.